src/lib/env.js

```diff
@@ -55,29 +67,44 @@ export class Env {
     const cache = this._resCache;
     const id = res + lang.code + lang.src;
 
-    if (cache[id]) {
-      return cache[id];
+    if (cache.has(id)) {
+      return cache.get(id);
     }
 
     const syntax = res.substr(res.lastIndexOf('.') + 1);
 
     const saveEntries = data => {
       const entries = this._parse(syntax, lang, data);
-      cache[id] = this._create(lang, entries);
+      cache.set(id, this._create(lang, entries));
     };
 
     const recover = err => {
       err.lang = lang;
       this.emit('fetcherror', err);
-      cache[id] = err;
+      cache.set(id, err);
     };
 
     const langToFetch = lang.src === 'pseudo' ?
       { code: this.defaultLang, src: 'app' } :
       lang;
 
-    return cache[id] = this.fetch(res, langToFetch).then(
+    const resource = this.fetch(res, langToFetch).then(
       saveEntries, recover);
+
+    cache.set(id, resource);
+
+    return resource;
+  }
+}
+
+function deleteIfOrphan(cache, lists, resId) {
+  let isNeeded = false;
+  lists.forEach(
+    resIds => isNeeded = resIds.has(resId) ? true : isNeeded);
```

Comment:
It would be possibly nice to stop looping on first `true`. Is there `Map` equivalent of `Array.some`?